src/main/java/com/networknt/schema/PropertiesValidator.java

```diff
@@ -135,7 +135,7 @@ private void applyPropertyDefaults(ObjectNode node) {
             if (propertyNode == null || (applyDefaultsStrategy.shouldApplyPropertyDefaultsIfNull() && propertyNode.isNull())) {
                 JsonSchema propertySchema = entry.getValue();
                 JsonNode defaultNode = propertySchema.getSchemaNode().get("default");
-                if (defaultNode != null && !defaultNode.isNull()) {
+                if (defaultNode != null) {
                     // mutate the input json
                     node.set(entry.getKey(), defaultNode);
                 }

```

Comment:
In ItemsValidator they have similar code
```
                if (defaultNode != null && defaultNode.isNull()) {
                    defaultNode = null;
                }
```
Maybe this should be removed?

Reply:
Can you add unit test to JsonWalkApplyDefaultsTest?  You'll have to change walk-schema-default.json, adding `"default": null` somewhere, and similar changes to walk-data-default.json.

Reply:
> In ItemsValidator they have similar code
> 
> ```
>                 if (defaultNode != null && defaultNode.isNull()) {
>                     defaultNode = null;
>                 }
> ```
> 
> Maybe this should be removed?

In ItemsValidator, this code is fine as Library specification says `applyArrayDefaults if true then apply defaults inside json arrays if the attribute is explicitly null` so code in here is checking if incoming item is null and default is not null then set that not null item in the array in place of null as there is no point of setting null in place of null. This is depicted in this code:

```
for (JsonNode n : arrayNode) {
                if (n.isNull() && defaultNode != null) {
                    arrayNode.set(i, defaultNode);
                    n = defaultNode;
                }
                doWalk(validationMessages, i, n, rootNode, at, shouldValidateSchema);
                i++;
            }
```

Reply:
> Can you add unit test to JsonWalkApplyDefaultsTest? You'll have to change walk-schema-default.json, adding `"default": null` somewhere, and similar changes to walk-data-default.json.

I have added some required test cases, Please have a look.

Reply:
> In ItemsValidator, this code is fine as Library specification says `applyArrayDefaults if true then apply defaults inside json arrays if the attribute is explicitly null` so code in here is checking if incoming item is null and default is not null then set that not null item in the array in place of null as there is no point of setting null in place of null. This is depicted in this code:

This makes sense.  However at some point I was thinking we may want to add an event listener, that's why I created the ApplyDefaultsStrategy class to add more stuff in the future if we want to.  Now we may want the listener to count the number of places where the default is missing or something else, so replacing null with null and firing the listener makes sense in that case.  What do you think?